File: 5/05698.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
	while (1) {
		string letter; getline(cin, letter);
		if (letter == "*") break;

		bool isTautogram = true;
		char alpha = toupper(letter[0]);
		for (int i = 1; i < letter.length(); i++)
			if (letter[i] == ' ')
				if (toupper(letter[i + 1]) != alpha)
					isTautogram = false;

		if (isTautogram) cout << "Y\n";
		else cout << "N\n";
	}
}

File: 5/05698.py
while 1:
    letter = list(input().split())
    if letter[0] == "*": break

    isTautogram = True
    alpha = letter[0][0].upper()
    for l in letter[1:]:
        if l[0].upper() != alpha:
            isTautogram = False

    if isTautogram: print("Y")
    else: print("N")
